apps/editors/helpers.py

```diff
@@ -44,12 +44,17 @@ def file_compare(file_obj, version):
 
 @register.function
 def file_review_status(addon, file):
-    if file.status not in [amo.STATUS_DISABLED, amo.STATUS_PUBLIC]:
+    # If the file is pending review, check the add-on status
+    if file.status == amo.STATUS_UNREVIEWED:
+        if addon.status in [amo.STATUS_NOMINATED, amo.STATUS_PUBLIC]:
+            return _(u'Pending Full Review')
         if addon.status in [amo.STATUS_UNREVIEWED, amo.STATUS_LITE]:
             return _(u'Pending Preliminary Review')
-        elif addon.status in [amo.STATUS_NOMINATED,
-                              amo.STATUS_LITE_AND_NOMINATED,
-                              amo.STATUS_PUBLIC]:
+    # Special case: prelim upgrading to full approval,
+    # file can already be preliminary reviewed or not
+    if (file.status in [amo.STATUS_LITE, amo.STATUS_UNREVIEWED]
```

Comment:
I know I repeat `amo.STATUS_UNREVIEWED` here but this is the cleanest way I could up with without making things hard to read.